### PR TITLE
YAML support

### PR DESCRIPTION
**Pull request by ckulka**
Seeing how YAML (<http://yaml.org/>) is easier for us humans, I added YAML support to the plugin. I also added jUnit tests to make sure there are no regressions.

My approach was to have as little impact on the existing code as possible and therefore simply read the YAML files, translate them to JSON and let the rest of the code handle it as before.

Let me know what you think, if I missed something, or should add to this PR.

### CODE REVIEW

**Comment by ssvaidyanathan** [comment]
HI @ckulka - Appreciate this PR. Have not looked into the code as yet.
The main reason why we used JSON over YAML is - the structure we used within the JSON configs uses the same payload used by [Apigee Management API](https://apidocs.apigee.com/management/apis). We wanted to make it easy for the users to just take the payload from our docs and paste it in the config files and use the plugin. With the YAML way, the developers need to be super careful about the structure and how they setup their configurations. I just feel it might add more work for the users (to convert the JSON payload to YAML). 
@msadasivam - your thoughts ?

**Comment by ckulka** [comment]
Sure, I can see that, especially when users are just getting started and just want to manage what they already created - which is why the code prefers JSON over YAML files.

My intended audience were those who are already there and want to take the next step, i.e. adding/modifying API Products, Developer Apps etc. directly from the repository without manually creating & exporting them from the UI. Those users that already made the transition would benefit from YAML, as it's easier for them to create/maintain their resources.

**TL;DR**
In other words, my rationale is similar to AWS CloudFormation and Kubernetes adding support for YAML in addition to JSON.

**Comment by ssvaidyanathan** [comment]
Agree but teams that have been using this plugin have been using JSON as they are using the Mgmt API payload directly. For any changes, JSON isnt that difficult as well. Though I agree YAML is way more easier (no brainer). Would be nice if we had a util that could convert the JSON to YAML so that teams can start using YAML or move over to YAML. Asking them to transition over to YAML manually might be not be motivating change for them :)

**Comment by msadasivam** [comment]
Appreciate the thought and PR. We use this plugin in the field a lot and this is the first time I have heard of this enhancement. I like the idea of supporting YAML but this may be too early for the benefit of the plugin. 

Can we do pre-processor conversion of YAML to JSON in a CI/CD tool to achieve the required result.

**Comment by ckulka** [comment]
Hi all,

since I was struggling a bit the additional code for YAML parsing and given YAML is a superset of JSON, I consolidated the parsers in the `ConfigReader.java` and `ConsolidatedConfigReader.java`.

The result is pretty much

- one library to read and parse configuration files (Jackson replacing the previous one)
- added YAML support
- arguably easier to read due to the API of the Jackson library

**Comment by ssvaidyanathan** [comment]
closing this issue as part of housekeeping activities
